storey/targets.py

```
@@ -686,10 +691,10 @@ async def _worker(self):
             await self._storage.close()
 
     async def _lazy_init(self):
-        if not self._shard_count:
-            response = await self._storage._describe(self._container, self._stream_path)
 
-            self._shard_count = response.shard_count
+        await self._storage._create_stream(self._container, self._stream_path, self._shard_count, self._retention_period_hours)
```

We don't want to call this every time we get an event.

---

Also, won't this fail if the stream already exists?

---

it ignores the error if stream already exists 
 
`res.raise_for_status([409, 204])`

---

Right, `raise_for_status` method name is misleading 🙈